linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/JSGen.scala

```diff
@@ -98,11 +98,36 @@ private[emitter] final class JSGen(val semantics: Semantics,
       VarDef(name, rhs = None)
   }
 
+  def genSelect(receiver: Tree, cls: ClassRef, field: irt.Ident)(
+      implicit pos: Position): Tree = {
+    genSelect(receiver, cls.className, field)
+  }
+
+  def genSelect(receiver: Tree, cls: String, field: irt.Ident)(
+      implicit pos: Position): Tree = {
+    DotSelect(receiver, genFieldIdent(cls, field)(field.pos))
+  }
+
+  private def genFieldIdent(cls: String, field: irt.Ident)(
+      implicit pos: Position): Ident = {
+    Ident(cls + "__f_" + field.name, field.originalName)
+  }
+
   def genSelectStatic(className: String, item: irt.Ident)(
       implicit pos: Position): VarRef = {
     envField("t", className + "__" + item.name)
   }
 
+  def genJSPrivateSelect(receiver: Tree, cls: ClassRef, field: irt.Ident)(
+      implicit pos: Position): Tree = {
+    DotSelect(receiver, genJSPrivateFieldIdent(cls.className, field)(field.pos))
+  }
+
+  def genJSPrivateFieldIdent(cls: String, field: irt.Ident)(
+      implicit pos: Position): Ident = {
+    Ident(cls + "__f_" + field.name, field.originalName)
+  }
+
```

Comment:
This is all the same than `genSelect`. So I assume this is forward proofing to be able to replace these things with symbols so that they can be hidden from users entirely. However, I wonder if we shouldn't do this type based instead.

Reply:
> So I assume this is forward proofing to be able to replace these things with symbols so that they can be hidden from users entirely.

That's right.

> However, I wonder if we shouldn't do this type based instead.

Hum, I don't understand what you mean. How would we do it type-based? Or *what* exactly would we do type-based?

Reply:
Seems to pretty much be the same story than above. Do this based on the class kind of `cls`. If it is a Scala class, put a field like now. If it is a JS class, do the fancy symbol (or whatever other thing).